### PR TITLE
Wraith fixes

### LINKED ISSUE #8648: Wraith can end phase shift inside walls
### Description

read title

### Test Merges

doesn't matter

### Reproduction Steps

1. Phase Shift
2. Unshift
3. You're inside wall now.

### Screenshots

I'm too lazy to shorten the video, cope.

https://user-images.githubusercontent.com/24631139/138552288-59d90730-77ac-4d53-b17e-d2e43fffb5c1.mov

### LINKED ISSUE #9045: Banishing a tunnel while a xeno climbs in allows them to climb out into banishspace
### Description

Read title

### Test Merges

Doesn't matter

### Reproduction Steps

1. Make 2 tunnels
2. Xeno crawls to tunnel B from tunnel A
3. Tunnel B is banished while the Xeno crawls towards it
4. They climb out the tunnel before banish ends
5. They're now in banishspace allowing them to interact with any humans banished inside wether it be facehugging or stinging

### Screenshots

I'm too lazy to record this

### LINKED ISSUE #9137: Wraith can regress while injured right after shifting due to how health updates
### Description

read title

### Test Merges

doesn't matter

### Reproduction Steps

1. Be hurt
2. Shift
3. Unshift
4. Regress before your health updates

### Screenshots

https://user-images.githubusercontent.com/24631139/147868206-a50b01b0-286d-48b2-91dc-424dc51ccf43.mp4

### LINKED ISSUE #9364: wraith banish can permanently send mobs to nullspace
### Description

what should happen: you get banished and come back
what does happen: you never come back

### Test Merges

Server-Side Code Modifications: https://tgstation13.org/parsed-logs/sybil/server_side_modifications.dm

Round ID: [13017](https://scrubby.melonmesa.com/round/13017)
BYOND Version: 514.1575
Server revision compiled on: 
Master commit: b7064e8ce6e45927602c9947b86f796203eca9fb
The following pull requests are currently test merged:
#9355: 'Debug pr to find out where that damn sleep is' by BraveMole at commit e6dc9a479e
#9333: 'Fix some move issues' by BraveMole at commit 07660a94d2
#9266: 'Pain update (tm first)' by BraveMole at commit e7fc7ada70
#8932: 'Icarus Shipyard. (New map)' by vilereaver41 at commit c49d622ff9
#9239: 'Adds the 6E Chameleon Cape' by hyper2snyper at commit 7148efa111
#9207: 'Add teleporters pad to req' by BraveMole at commit bed7f685a7
#9227: 'Evacuation can prevent xeno major' by BraveMole at commit 1f287382ed

Local commit: 7e39a8254b2727d6f5dc0be7cacdb3aff3829c74
Server tools version: /tg/station 13 Server v3.2.6.2

### Reproduction Steps

1. banish a mob standing under where a dropship will land (any tile that would normally gib the mob when the ds lands)
2. dropship lands, therefore technically deleting the turf that you were banished from
3. now that the turf it was banished from doesn't exist, there is no place to send it back to and it is in nullspace forever
this probably works backwards (with the banished mob being ON a DS that is about to take off)

### Screenshots

_No response_

### LINKED ISSUE #9466: Wraiths can't banish airlocks or shutters
### Description

read title, Line of Sight changes did this.

### Test Merges

doesn't matter

### Reproduction Steps

1. Walk up to door/shutter
2. Try to banish
3. Cope

### Screenshots

https://user-images.githubusercontent.com/24631139/151729358-3d1677d5-c7a5-490d-98c0-5e372ff91a18.mp4

### LINKED ISSUE #9680: Wraiths can vanish "Hull Window" on Prison Station
### Description

title, able to go out and swim in the sea and die, so sad can we hit 50 PRs closed?

### Test Merges

not relevant

### Reproduction Steps

Be wraith
Select banish
Find Hull Window
???
Don't profit

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by breadhunt**
## About The Pull Request

Lot of wraith fixes
Stuff I couldn't reproduce/seemed to be already fixed:
-timestop not working https://github.com/tgstation/TerraGov-Marine-Corps/issues/9859
-wraiths can't unbanish themselves https://github.com/tgstation/TerraGov-Marine-Corps/issues/9690
-Banished xenos taking fire damage https://github.com/tgstation/TerraGov-Marine-Corps/issues/9136
-Blinking marines past fog (the fog just stops you) https://github.com/tgstation/TerraGov-Marine-Corps/issues/7268

What is fixed:
- Closes #9045
- Closes #9680
- Closes #9466
- Closes #9364
- Closes #9137
- Closes #8648
- Tadpole can no longer be slashed **while in phase shift** (this was extremely funny)
- Starting/ending phase shift in no-phase zones and objects
^ This means you can no longer accidentally get stuck in no-phase shutters, you just get teleported out

Alamo/canter fuckery is still very possible but since you can't use phase shift to get in (shifting into walls, shifting into cockpit) it might be a bit harder (probably for the best)

I don't know how to fix timestop bugs. If someone with better c*ding socks can figure them out here's a (probably) list of things to check out:
- Lasers can completely crash the game if spammed into timestop
- Lag means bullets travel further before being stopped by timestop (sometimes making it completely worthless)
- Bullets probably shouldn't be able to be fired at all in timestop considering it's a timestop

## Why It's Good For The Game

Bugfixes good. Wraith has a bad rep for a few REALLY bad bugs that haven't been fixed for too long. Fixes them and might Make Wraith Playable Again:tm:

## Changelog
:cl:
fix: Wraiths can no longer banish tunnels, deleting any xenos climbing inside
fix: Prison Station hull windows can no longer be banished, allowing access to space
fix: Wraiths can no longer use phase shift in no-phase areas (e.g LZs, dropships, spess)
fix: Wraiths can no longer end phase shift, or have phase shift ended, in no-phase areas or dense objects (walls)
fix: This means your buddy wraith won't get stuck in a phase shutter in a wall by accident again
fix: Wraith can no longer slash tadpole console while in phase shift
fix: Wraiths can banish opaque objects (shutters, doors) again
fix: Wraith can no longer regress in phase shift (caused issues because of how health updates work with phase shift, sorry)
fix: Banish portal being deleted (by a shuttle, etc), or otherwise made impossible to move to, no longer keeps the marine in banish indefinitely. Instead they are teleported to the coordinates the portal was originally at
/:cl:

### CODE REVIEW

**Comment by Wayland-Smithy** [comment]
Very cool. Unfortunately though all those issues will not auto close on merge because github typically parses for "Fixes: #xxxx" or "Closes: #xxxx" and I'm not sure just adding the prefix will work with the nice named links, read more about it [here](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue). You'll now it's working when the issues are listed in the Development section on the right side.

**Comment by breadhunt** [comment]
> Very cool. Unfortunately though all those issues will not auto close on merge because github typically parses for "Fixes: #xxxx" or "Closes: #xxxx" and I'm not sure just adding the prefix will work with the nice named links, read more about it [here](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue). You'll know it's working when the issues are listed in the Development section on the right side.

GOD BLESS YOU WAYLAND.
Done

**Comment by BraveMole** [comment]
Next time please atomize, it is pain to review this

File: code/modules/shuttle/mini_dropship.dm [hunk 1]
```diff
@@ -152,6 +152,8 @@
 	. = ..()
 	if(damaged)
 		return
+	if(isxenowraith(X) && X.status_flags & INCORPOREAL)
```
Comment (BraveMole) [inline code/modules/shuttle/mini_dropship.dm:155]: Istype(src) not allowed

Comment (BraveMole) [reply code/modules/shuttle/mini_dropship.dm:155]: But you probaboy just needs the status flag anyway

File: code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm [hunk 2]
```diff
@@ -546,6 +562,10 @@ GLOBAL_LIST_INIT(wraith_banish_very_short_duration_list, typecacheof(list(
 	var/atom/movable/banishment_target = null
 	///SFX indicating the banished target's position
 	var/obj/effect/temp_visual/banishment_portal/portal = null
+	///Backup coordinates to teleport the banished to, in case the portal gets destroyed (shuttles!!)
+	var/backup_x
+	var/backup_y
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm:567]: The fix idea looks correct, please save the coordinates into an array

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm [hunk 3]
```diff
@@ -664,8 +685,11 @@ GLOBAL_LIST_INIT(wraith_banish_very_short_duration_list, typecacheof(list(
 	SIGNAL_HANDLER
 	if(!banishment_target)
 		return
+	if(get_turf(portal))
+		banishment_target.forceMove(get_turf(portal))
+	else //The portal got removed. There's no way back. Ohgodohfuck.
+		banishment_target.forceMove(locate(backup_coordinates[1],backup_coordinates[2],backup_coordinates[3]))
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm:691]:
```suggestion
		banishment_target.forceMove(locate(backup_coordinates[1], backup_coordinates[2], backup_coordinates[3]))
```

**Review by BraveMole — APPROVED** [review]

**Pull request merged**